core/generator.js

```diff
@@ -77,6 +77,12 @@ Blockly.Generator.prototype.COMMENT_WRAP = 60;
  */
 Blockly.Generator.prototype.ORDER_OVERRIDES = [];
 
+/**
+ * Whether the init method has been called.
```

Comment:
I'm not sure if there's additional information I could add to help clarify how this is used.

Our core custom generators set this to `false` after creation of the singleton and then to `true` in `init`. 
`blockToCode` uses this to emit a helpful warning only if it has been set to `false`.

Reply:
I think adding what you have is good but maybe change: only if it has been set to false --> when the generator has not been initialized.

Reply:
Updated as per discussion offline.